lib/tiled/tiled_object.rb

```diff
@@ -47,6 +51,10 @@ def initialize(map, attrs, children)
       end
     end
 
+    def properties
+      @properties ||= Properties.new(self)
```

Comment:
You need to pass `map` instead of `self`, I think you will get an error if an object's property will have the `object` type.

Reply:
Good catch- just instantiating it didn't throw an error but that was bound to cause pain down the line.

Reply:
yep. I dream of a map example with all crazy features that Tiled supports to catch such things as early as possible :)

Reply:
Haha, I just spent like 5 hours installing RSpec in my library for that exact benefit :sweat_smile: It was a bitch and a half but it'll be really nice for [pixel-perfect behavior testing](https://github.com/vinnydiehl/drtiled-renderer/blob/main/spec/tiled_renderer/camera_spec.rb#L132-L167). It's a pipe dream of mine to get RSpec running in the DragonRuby runtime (easy, right? just drop it into your `lib/` and then fix all the errors), but for now, this hack job will do for what is essentially just model testing.

Reply:
With dream map it will be easy to write system spec:

```ruby
Rspec.describe Tiled::Map do
  it 'works' do
    expect { Tiled::Map.new('dream_map.tmx').load }.not to raise_error
  end
end
```

:D